test/test_examples.jl

```diff
@@ -0,0 +1,39 @@
+EXAMPLES_DIR = "../examples/"
+
+function run_deepening_mixed_layer_example(arch)
+    example_filepath = joinpath(EXAMPLES_DIR, "deepening_mixed_layer.jl")
+    txt = read(example_filepath, String)
+
+    arch == GPU() && (txt = replace(txt, "arch = CPU()" => "arch = GPU()"))
+
+    txt = replace(txt, "N = 32" => "N = 16")
```

Comment:
Fine for now, but we should write a function that auto-changes example parameters so we don't have to copy-paste these lines for each example.